module-4/nmap/nmap.py

```diff
@@ -0,0 +1,80 @@
+import socket
+import argparse
+import concurrent.futures
+
+# Common services for the first 100 ports
+COMMON_PORTS = {
+    20: "FTP data",
+    21: "FTP control",
+    22: "SSH",
+    23: "Telnet",
+    25: "SMTP",
+    53: "DNS",
+    80: "HTTP",
+    110: "POP3",
+    143: "IMAP",
+    443: "HTTPS",
+    993: "IMAPS",
+    995: "POP3S",
+    # ... continue up to port 100 as needed
+}
+
+def resolve_host(host):
+    try:
+        ip = socket.gethostbyname(host)
+        print(f"[INFO] IP address for {host} is {ip}")
+        return ip
+    except socket.gaierror:
+        print(f"[ERROR] Could not resolve host {host}.")
+        return None
+
+def scan_port(ip, port):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.settimeout(1)
+        result = s.connect_ex((ip, port))
+        if result == 0:
+            service_name = COMMON_PORTS.get(port, "Unknown Service") if port <= 100 else "Unknown"
+            print(f"Port {port} is open (Service: {service_name})")
+            return port, True, service_name
+        else:
+            print(f"Port {port} is closed")
+            return port, False, None
+
+def port_scan(ip, port_range):
+    open_ports = []
+    with concurrent.futures.ThreadPoolExecutor(max_workers=100) as executor:
+        future_to_port = {executor.submit(scan_port, ip, port): port for port in port_range}
```

Comment:
This name is not quite good. The problem is:

- I cannot tell what this variable will contain as value
- I cannot determine what the variable is used for semantically